Source/DafnyPipeline.Test/ProverLogStability.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using DafnyCore.Test;
+using DafnyTestGeneration;
+using Microsoft.Boogie;
+using Microsoft.Dafny;
+using Xunit;
+using Xunit.Abstractions;
+using BoogieProgram = Microsoft.Boogie.Program;
+
+namespace DafnyPipeline.Test {
+  // Main.Resolve has static shared state (TypeConstraint.ErrorsToBeReported for example)
+  // so we can't execute tests that use it in parallel.
+  [Collection("Singleton Test Collection - Resolution")]
+  public class ProverLogStabilityTest {
+
+    private readonly ITestOutputHelper testOutputHelper;
+
+    // All types of top level declarations.
+    readonly string originalProgram = @"
+module SomeModule {
+
+  module NestedModule {
+    class C {
+      var f: int
+      constructor ()
+    }
+  }
+
+  method m() {
+    var x: NestedModule.C;
+    x := new NestedModule.C();
+    x.f := 4;
+  }
+}
+
+import opened SomeModule
+
+type FooSynonym<T> = FooClass
+
+class FooClass {
+  var f: int
+  constructor ()
+}
+
+datatype Friends = Agnes | Agatha | Jermaine
+
+function SomeFunc(funcFormal: int): nat { 3 }
+
+method SomeMethod(methodFormal: int) returns (result: bool)
+  requires methodFormal == 2
+  ensures result == true 
+  // ensures forall x :: x == methodFormal
+{
+  m();
+  var lambdaExpr := x => x + 1;
+  result := methodFormal == SomeFunc(42);
+}
+";
+
+    public ProverLogStabilityTest(ITestOutputHelper testOutputHelper) {
+      this.testOutputHelper = testOutputHelper;
+    }
+
+    [Fact]
+    public async Task ProverLogRegression() {
```

Comment:
I think we need a sizable comment here to describe the purpose of the test. Something like "this test only provides data on when changes are made that change the encoding of Dafny programs to SMT programs. A failure is meant to provide this data, not block changes - if you are contributing a change that makes it fail...". Probably even with some background about the fundamental brittleness problem.